### PR TITLE
[#9079] Migrate all admin pages and tidy up UI design

### LINKED ISSUE #8421: Admin: ongoing sessions list: fix missing sessions
As discovered by @tran-tien-dat ([link](https://github.com/TEAMMATES/teammates/commit/c48ed139c973330b86d3c4e1f43f760cf4bf9c1b#r27445177)), sessions that start before the specified "from" date and end after the specified "to" date are not shown under ongoing sessions even though they are open.

### LINKED ISSUE #9079: Set up foundation for front-end migration project
To kick off [this project](https://github.com/TEAMMATES/teammates/projects/4), some foundations need to be laid out first:
- [x] Step 1: Set up Angular 6 front-end project
  - Included: lint/build/test routine (including reliable code coverage), separate dev server from the back end, hot reload in development mode, source maps, minification when building for production usage
- [x] Step 2: Set up REST-ful back end servlet, ~possibly with the help of [Jersey](https://medium.com/@tejohnso/a-minimal-java-api-with-jetty-jersey-jackson-and-gradle-9842a145815e),~ and user authentication workflow
- [x] Step 3: Migrate few pages; concurrently, set up example of how migration for each page should be done systematically.

- [ ] Step 4 (can be deferred): Set up example tests, ranging from front-end component tests (with Karma/Jasmine), API endpoint tests (with TestNG), and E2E tests (with Selenium).

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #9079 (this time for real!)

- [x] Few more infrastructure improvements
- [x] Tidied up UI design issues, or most of it
- [x] Migrated admin sessions page and admin account details page
  - [x] Fixed the query for ongoing sessions, so this also fixed #8421 (see 6422f3b)
  - [x] Added support for resetting Google ID for instructors, which is a partial fix for #7062
- [x] Decommissioned admin activity log page, admin email log page, and admin account management page (all the relevant functions of admin account management page has been ported somewhere else)
  - This accounts for the net deletion of ~5,600 lines of code.
- [x] Decommissioned admin email features
  - This accounts for another net deletion of ~4,600 lines of code.

### CODE REVIEW

File: src/web/services/http-request.service.ts [hunk 1]
```diff
@@ -21,10 +21,12 @@ export class HttpRequestService {
   /**
    * Builds an HttpParams object from a standard key-value mapping.
    */
-  buildParams(paramsMap: any): HttpParams {
+  buildParams(paramsMap: { [key: string]: string }): HttpParams {
     let params: HttpParams = new HttpParams();
     for (const key of Object.keys(paramsMap)) {
-      params = params.append(key, paramsMap[key]);
+      if (paramsMap[key]) {
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/services/http-request.service.ts]: Take a look at the type of `paramsMap`: it's a string-string mapping. `paramsMap[key]` is a guard for empty string, not for `null`/`undefined`.

If you want to pass `false`, then pass the string `"false"`.

File: src/web/app/pages-instructor/instructor-page.component.ts [hunk 2]
```diff
@@ -53,7 +56,10 @@ export class InstructorPageComponent implements OnInit {
           this.logoutUrl = `${this.backendUrl}${res.logoutUrl}`;
         }
         if (res.user) {
-          this.isValidUser = res.user.isInstructor;
+          this.user = res.user.id + (res.masquerade ? ' (M)' : '');
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/app/pages-instructor/instructor-page.component.ts]: Should be doable.

File: src/main/java/teammates/ui/newcontroller/GetOngoingSessionsAction.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+                return logic.getAccount(instructor.googleId);
+            }
+        }
+        return null;
+    }
+
+    private static class OngoingSession {
+
+        private final String sessionStatus;
+        private final String instructorHomePageLink;
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/ui/newcontroller/GetOngoingSessionsAction.java]: Haven't explored that option yet. Will do some time later.

File: src/web/app/pages-admin/admin-sessions-page/admin-sessions-page.component.ts [hunk 4]
```diff
@@ -10,9 +13,130 @@ import { Component, OnInit } from '@angular/core';
...
+    inst.set('hour', hour);
+    inst.set('minute', minute);
+    return inst;
+  }
+
+  /**
+   * Gets the feedback sessions which have opening time satisfying the query range.
+   */
+  getFeedbackSessions(): void {
+    const startTime: any = this.getMomentInstant(this.startTimeDate.year, this.startTimeDate.month - 1,
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/app/pages-admin/admin-sessions-page/admin-sessions-page.component.ts]: `getMomentInstant` already handles that.

File: src/web/app/pages-admin/admin-accounts-page/admin-accounts-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    });
+  }
+
+  /**
+   * Removes the instructor from course.
+   */
+  removeInstructorFromCourse(courseId: string): void {
+    const id: string = this.accountInfo.googleId;
+    const paramMap: { [key: string]: string } = {
+      googleid: id,
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/app/pages-admin/admin-accounts-page/admin-accounts-page.component.ts]: Can't believe I missed them.

> You may swap the parameters of the two methods removeStudentFromCourse and removeInstructorFromCourse accidentally.

No, you can't; the back-end endpoints should not accept wrong param names.

File: src/main/java/teammates/ui/newcontroller/ResetAccountAction.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        String wrongGoogleId = null;
+        if (studentEmail != null) {
+            StudentAttributes existingStudent = logic.getStudentForEmail(courseId, studentEmail);
+            wrongGoogleId = existingStudent.googleId;
+
+            try {
+                logic.resetStudentGoogleId(studentEmail, courseId);
+                taskQueuer.scheduleCourseRegistrationInviteToStudent(courseId, studentEmail, true);
+            } catch (EntityDoesNotExistException | InvalidParametersException e) {
+                // TODO
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/ui/newcontroller/ResetAccountAction.java]: Can't say I'm sure yet. I'm still deciding between just throwing an unchecked exception here or changing the logic layer to not throw those exceptions.

**Comment by wkurniawan07** [comment]
@xpdavid I'm not done yet, but can you review the new changes pushed first?

**Comment by wkurniawan07** [comment]
@jacoblipech, looks like you accidentally deleted the `teammatesv7` branch?

**Comment by jacoblipech** [comment]
@wkurniawan07 omg my bad!!! I was looking through and deleted it by accident!

File: src/web/app/pages-admin/admin-sessions-page/admin-sessions-page.component.ts [hunk 7]
```diff
@@ -1,4 +1,34 @@
...
+  instructorHomePageLink: string;
+  startTime: number;
+  endTime: number;
+  creatorEmail: string;
+  courseId: string;
+  feedbackSessionName: string;
+  responseRate?: string;
+}
+
+interface OngoingSessionsData {
```
Comment (xpdavid) [inline src/web/app/pages-admin/admin-sessions-page/admin-sessions-page.component.ts]: OngoingSessionsStats?

File: src/web/app/pages-static/about-page/about-page.component.html [hunk 8]
```diff
@@ -1,137 +1,125 @@
...
+      <b>Google Summer of Code</b> Program, for including TEAMMATES as a mentor organization in GSoC2014, GSoC2015, GSoC2016, GSoC2017, and GSoC2018 editions.
+    </li>
+    <li>
+      <b>Facebook Open Academy</b> Program, for including TEAMMATES as a mentor organization in FBOA 2016.
+    </li>
+    <li>
+      <b>Jet Brains</b>, for the <a href="https://www.jetbrains.com/idea/" target="_blank" rel="noopener noreferrer">Intellij IDEA</a> licences.
+    </li>
+    <li>
+      <b>YourKit LLC</b>, for providing us with free licenses for the <a href="https://www.yourkit.com/java/profiler/" target="_blank" rel="noopener noreferrer">YourKit Java Profiler</a>
```
Comment (xpdavid) [inline src/web/app/pages-static/about-page/about-page.component.html]: Don't forget SauceLabs when resolve conflict :P

File: src/main/java/teammates/ui/newcontroller/DeleteInstructorAction.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (userInfo.isAdmin) {
+            return;
+        }
+        // TODO allow access to instructors with modify permission
```
Comment (xpdavid) [inline src/main/java/teammates/ui/newcontroller/DeleteInstructorAction.java]: Just to take note. We need to enforce the constraint that at least one instructor should be in the course.

**Pull request merged**